src/Signup_Command.php

```diff
@@ -0,0 +1,272 @@
+
+		if ( 'ids' === $format ) {
+			WP_CLI::line( implode( ' ', wp_list_pluck( $signups, 'signup_id' ) ) );
+		} else {
+			$formatter->display_items( $signups );
+		}
+	}
+
+	/**
+	 * Gets details about a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : The signup ID, user login, user email, or activation key.
+	 *
+	 * [--field=<field>]
+	 * : Instead of returning the whole signup, returns the value of a single field.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields. Defaults to all fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get signup.
+	 *     $ wp user signup get 1 --field=user_login
+	 *     bobuser
+	 *
+	 *     # Get signup and export to JSON file.
+	 *     $ wp user signup get bobuser --format=json > bobuser.json
+	 *
+	 * @package wp-cli
+	 */
+	public function get( $args, $assoc_args ) {
+		$signup = $this->fetcher->get_check( $args[0] );
+
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( (array) $signup );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		$formatter->display_items( array( $signup ) );
+	}
+
+	/**
+	 * Activates one or more signups.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>...
+	 * : The signup ID, user login, user email, or activation key of the signup(s) to activate.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Activate signup.
+	 *     $ wp user signup activate 2
+	 *     Success: Signup 2 activated. Password: bZFSGsfzb9xs
+	 *
+	 * @package wp-cli
+	 */
+	public function activate( $args, $assoc_args ) {
+		$signups = $this->fetcher->get_many( $args );
+
+		foreach ( $signups as $signup ) {
+			$result = wpmu_activate_signup( $signup->activation_key );
+
+			if ( is_wp_error( $result ) ) {
+				WP_CLI::warning( "Failed activating signup {$signup->signup_id}." );
+			} else {
+				WP_CLI::success( "Signup {$signup->signup_id} activated. Password: {$result['password']}" );
+			}
+		}
+	}
+
+	/**
+	 * Deletes one or more signups.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>...
+	 * : The signup ID, user login, user email, or activation key of the signup(s) to delete.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Delete signup.
+	 *     $ wp user signup delete 3
+	 *     Success: Signup 3 deleted.
+	 *
+	 * @package wp-cli
+	 */
+	public function delete( $args, $assoc_args ) {
+		$signups = $this->fetcher->get_many( $args );
+
+		parent::_delete( $signups, $assoc_args, [ $this, 'delete_callback' ] );
+	}
+
+	/**
+	 * Callback used to delete a signup.
+	 *
+	 * @param $signup
+	 * @param $assoc_args
+	 * @return array
+	 */
+	protected function delete_callback( $signup, $assoc_args ) {
+		global $wpdb;
+
+		$signup_id = $signup->signup_id;
+
+		$result = $wpdb->delete( $wpdb->signups, array( 'signup_id' => $signup_id ), array( '%d' ) );
```

Comment:
How does WordPress core handle deletion?

Also, is there a cache we need to clear?

Reply:
From what I can see there are no caches involved with signups. Deletion happens like this, see e.g. `wp_delete_signup_on_user_delete()`